### PR TITLE
Configure a secure Oracle NoSQL Database CE installation

### PR DESCRIPTION
**Pull request by dario-vega**
This PR updates with some improvements to the Oracle NoSQL CE image.
`create and configure a secure Oracle NoSQL Database CE installation`

References
- https://community.oracle.com/tech/developers/discussion/4496876

Signed-off-by: Dario Vega dario.vega@oracle.com

### CODE REVIEW

**Review by mikebrey — DISMISSED** [review]

File: .github/workflows/build-and-push-nosql-sec-image.yml
Comment (Djelibeybi) [inline .github/workflows/build-and-push-nosql-sec-image.yml:9]: This path doesn't match the path of this file.

File: .github/workflows/build-and-push-nosql-sec-image.yml
Comment (Djelibeybi) [inline .github/workflows/build-and-push-nosql-sec-image.yml:14]: These are the same `IMAGE_NAME` and `TAG` values as the existing NoSQL workflow. Please edit the existing file instead of creating a new one (or delete the old workflow).

File: .github/workflows/build-and-push-nosql-sec-image.yml
Comment (Djelibeybi) [inline .github/workflows/build-and-push-nosql-sec-image.yml:21]: This will limit the workflow to only trigger on `push` even though you have `paths` configured in the `on` section. If you only want to trigger a build on push to main, just remove the `paths` section from the `on` configuration above.

File: .github/workflows/build-and-push-nosql-sec-image.yml
Comment (Djelibeybi) [inline .github/workflows/build-and-push-nosql-sec-image.yml:52]: If you want to change the suffix to `-ce-sec`, do that in the configuration above, don't override it here.

File: NoSQL/README-sec.md
Comment (Djelibeybi) [inline NoSQL/README-sec.md:15]: Why would we want to continue hosting a non-secure version? Why not replace the existing image with this secure version instead? Oracle best practice for releasing sample/demo code would require this anyway.

**Review by Djelibeybi — CHANGES_REQUESTED** [review]
Why are we maintaining a non-secure image? I think you should replace the existing image with the secure one and just have a single, more secure image that aligns better with Oracle best practice.

File: .github/workflows/build-and-push-nosql-sec-image.yml
Comment (dario-vega) [reply .github/workflows/build-and-push-nosql-sec-image.yml:9]: I tested it and working on my fork, so for me the path is good

image

There is probably a misunderstanding here: there is no old workflow and a new one - we want to maintain two versions

File: .github/workflows/build-and-push-nosql-sec-image.yml
Comment (dario-vega) [reply .github/workflows/build-and-push-nosql-sec-image.yml:14]: I am rewriting the tag later in the code - I need to review and delete the TAG from here. 

There is probably a misunderstanding here: there is no old workflow and a new one - we want to maintain two versions

File: .github/workflows/build-and-push-nosql-sec-image.yml
Comment (dario-vega) [reply .github/workflows/build-and-push-nosql-sec-image.yml:21]: I didn't understand, I am reusing the previous script

File: .github/workflows/build-and-push-nosql-sec-image.yml
Comment (dario-vega) [reply .github/workflows/build-and-push-nosql-sec-image.yml:52]: We want the following TAGS
- `${{ steps.date.outputs.date }}-ce-sec`
- `latest-ce-sec`

File: NoSQL/README-sec.md
Comment (dario-vega) [reply NoSQL/README-sec.md:15]: @mikebrey comments? My understanding was that we wanted to maintain the two versions.
- Oracle NoSQL Database **secure** configuration
- Oracle NoSQL Database **non-secure** configuration

Comment (Djelibeybi) [reply NoSQL/README-sec.md:15]: Maintaining a non-secure version would violate more Oracle software assurance and release policies than I care to consider, so if this is your intention, I'd need to get sign off from your security lead and from other folks in release management before merging.

Comment (mikebrey) [reply NoSQL/README-sec.md:15]: @Djelibeybi  The docker container that we are releasing is KVLITE.  KVLITE is a single process version of NoSQL for testing purposes only.     It is specifically designed for a single user.   With that being said, it can be configured in a secure or a non-secure mode similar to the NoSQL product itself.   We got feedback from several customers wanting to use KVLITE as a way to understand how to set up the real database in a secure manner.    Within docker we provide the scripts that get executed to set up KVLITE in a secure manner.  The concepts in those scripts can be used to set up the actual NoSQL database in a secure manner.    For the product itself, it complies with the security guidelines and we release with the defaults for a secure configuration.    KVLITE, as a testing tool, is a way for customers to test their application without having to set up a real database.   KVLITE is not the product.   If based on this, you still feel the need to get security lead sign off, I can pursue that.

Comment (Djelibeybi) [reply NoSQL/README-sec.md:15]: Oracle policy around security is generally pretty strict in this regard, and given the use of the terms "secure" and "non-secure" to describe each image intended for public release/use and because I'm not qualified to determine the security posture of KVLite in any configuration, I'd prefer to get sign-off from your security lead to make sure that even the non-secure image complies with the required Oracle security policy requirements for release, with the secure option adding additional security on top of the minimum requiement.

I've switched this PR to draft until we connect internally and the security review and sign-off is completed.

Comment (mikebrey) [reply NoSQL/README-sec.md:15]: @Djelibeybi  Avi - I understand the position you are taking.    I will discuss with our security lead in database.

**Comment by Djelibeybi** [comment]
@mikebrey any updates on this?

**Comment by mikebrey** [comment]
Not yet – expect to have something by next week.

From: Avi Miller ***@***.***>
Sent: Monday, October 3, 2022 4:29 PM
To: oracle/docker-images ***@***.***>
Cc: Michael Brey ***@***.***>; Mention ***@***.***>
Subject: [External] : Re: [oracle/docker-images] Configure a secure Oracle NoSQL Database CE installation (PR #2439)


@mikebrey<https://urldefense.com/v3/__https:/github.com/mikebrey__;!!ACWV5N9M2RV99hQ!J7k0ctHdxxHqJmZ4DIJxAlE3-GRL__dyhLty7HK4EM8FZJMgWDqpFvqqBphWywAG3hv_i4jHJYRa7e5k1mk6iSTGVCA$> any updates on this?

**Comment by dario-vega** [comment]
@mikebrey  can you provide the last update from your meeting with security team?

**Comment by mikebrey** [comment]
@Djelibeybi  and @dario-vega  I had a discussion with the database security lead on this.    Apologies for delay in getting this posted but last week I was at CloudWorld.     Security was ok with us having both secure and non-secure versions out there.   This is single process versions of the product.   Security wanted us to "push" the use of the secure version which we can do in the README.

**Comment by Djelibeybi** [comment]
Thanks @mikebrey. Could you please update this PR with the appropriate wording added to the README.md and then click "Ready for review".

**Comment by mikebrey** [comment]
@Djelibeybi   Yes - I will take care of that.

**Comment by dario-vega** [comment]
@Djelibeybi, Can we review it again?

**Comment by Djelibeybi** [comment]
There is a button next to my name that re-requests a review on GitHub. Please push that button so that this PR returns to my queue.

**Comment by Djelibeybi** [comment]
Please address the linting issues for both the Markdown files and shell scripts

**Review by Djelibeybi — CHANGES_REQUESTED** [review]
Please resolve the issues raised by the Markdown and shell script linters.

**Comment by dario-vega** [comment]
> Please resolve the issues raised by the Markdown and shell script linters.

done

**Review by Djelibeybi — APPROVED** [review]

**Comment by Djelibeybi** [comment]
Needs a code owner approval from @mikebrey after which it can be merged.

**Review by mikebrey — APPROVED** [review]

**Review by mikebrey — APPROVED** [review]

**Review by mikebrey — APPROVED** [review]

**Review by mikebrey — APPROVED** [review]

**Pull request merged**
